be/src/exec/parquet_writer.cpp

```diff
@@ -40,6 +40,124 @@ ParquetOutputStream::~ParquetOutputStream() {
+    switch (column_data_type) {
+    case TParquetDataType::BOOLEAN: {
+        parquet_data_type = parquet::Type::BOOLEAN;
+        break;
+    }
+    case TParquetDataType::INT32: {
+        parquet_data_type = parquet::Type::INT32;
+        break;
+    }
+    case TParquetDataType::INT64: {
+        parquet_data_type = parquet::Type::INT64;
+        break;
+    }
+    case TParquetDataType::INT96: {
+        parquet_data_type = parquet::Type::INT96;
+        break;
+    }
+    case TParquetDataType::BYTE_ARRAY: {
+        parquet_data_type = parquet::Type::BYTE_ARRAY;
+        break;
+    }
+    case TParquetDataType::FLOAT: {
+        parquet_data_type = parquet::Type::FLOAT;
+        break;
+    }
+    case TParquetDataType::DOUBLE: {
+        parquet_data_type = parquet::Type::DOUBLE;
+        break;
+    }
+    case TParquetDataType::FIXED_LEN_BYTE_ARRAY: {
+        parquet_data_type = parquet::Type::FIXED_LEN_BYTE_ARRAY;
+        break;
+    }
+    default:
+        parquet_data_type = parquet::Type::UNDEFINED;
+    }
+}
+
+void ParquetBuildHelper::build_compression_type(
+        parquet::WriterProperties::Builder& builder,
+        const TParquetCompressionType::type& compression_type) {
+    switch (compression_type) {
+    case TParquetCompressionType::SNAPPY: {
+        builder.compression(parquet::Compression::SNAPPY);
+        break;
+    }
+    case TParquetCompressionType::GZIP: {
+        builder.compression(parquet::Compression::GZIP);
+        break;
+    }
+    case TParquetCompressionType::BROTLI: {
+        builder.compression(parquet::Compression::BROTLI);
+        break;
+    }
+    case TParquetCompressionType::ZSTD: {
+        builder.compression(parquet::Compression::ZSTD);
+        break;
+    }
+    case TParquetCompressionType::LZ4: {
+        builder.compression(parquet::Compression::LZ4);
+        break;
+    }
+    case TParquetCompressionType::LZO: {
+        builder.compression(parquet::Compression::LZO);
+        break;
+    }
+    case TParquetCompressionType::BZ2: {
+        builder.compression(parquet::Compression::BZ2);
+        break;
+    }
+    case TParquetCompressionType::UNCOMPRESSED: {
+        builder.compression(parquet::Compression::UNCOMPRESSED);
+        break;
+    }
+    default:
+        builder.compression(parquet::Compression::UNCOMPRESSED);
+    }
+}
+
+void ParquetBuildHelper::build_version(parquet::WriterProperties::Builder& builder,
+                                       const TParquetVersion::type& parquet_version) {
+    switch (parquet_version) {
+    case TParquetVersion::PARQUET_1_0: {
+        builder.version(parquet::ParquetVersion::PARQUET_1_0);
+        break;
+    }
+    case TParquetVersion::PARQUET_2_LATEST: {
+        builder.version(parquet::ParquetVersion::PARQUET_2_LATEST);
+        break;
+    }
+    default:
+        builder.version(parquet::ParquetVersion::PARQUET_1_0);
```

Comment:
<img width="577" alt="image" src="https://user-images.githubusercontent.com/87313068/191435673-aa1baf79-15a6-4940-b5ab-1429e6ed354f.png">

before refactor, if haven't set version property, it will be PARQUET_1_0,